### ISSUE TITLE
Grid List Accessibility Issues

### ISSUE THREAD

**Issue by puru-hk** [SAP/fundamental-ngx#4666]
Grid List Accessibility Issues

Attached the file
[Grid List.docx](https://github.com/SAP/fundamental-ngx/files/5980066/Grid.List.docx)

**Comment by kavya-b** [comment]
- [ ] OB: on accessing the ‘Sort’ button, the table header details is not conveyed., highlighted in red.
EB: Ideally, all visually visible content should be conveyed to the screen reader user.
- [ ] OB: on accessing the card the second tab there is an additional information
EB: Ideally, the additional information must be removed that may be confusing for the user
- [ ] OB: on accessing the Cards, the tile count is not read out.
EB: Ideally, The list count must be read for the screen reader user
- [ ] OB: on accessing the “sort’ button, the table header details ‘ Products(7) is not read
EB:Ideally, the table header details must be read for the screen reader user
- [ ] OB:On accessing the ‘Sort’, ’Filter’, ’Group’, ’Navigation’ buttons, there is no action performed by using ‘Enter/Space Bar’ key.
EB: Ideally, the Button must be activated using Enter/Space Bar key.
**From Kavya: assuming that click/enter action is missing. Currently logs are placed in console, will replace these with alerts**

**Comment by Sanjay-1312** [comment]
Issue 1:
OB: Additional information is conveyed.
![image](https://user-images.githubusercontent.com/78714518/122026930-49407580-cde8-11eb-8071-3e0d993b8293.png)
EB: ideally, the table name Products(6) and the Sort button information should be read.

Issue 2:
Fixed 

issue 3:
Fixed

Issue 4:
![image](https://user-images.githubusercontent.com/78714518/122028004-37130700-cde9-11eb-8705-0b4593ee3d37.png)
OB: Additional information is conveyed.
EB: ideally, the table name Products(7) and the Sort button information should be read.

Issue 5:
Fixed

The cards are not navigable by arrow keys.

**Comment by kavya-b** [comment]
> Issue 1:
> OB: Additional information is conveyed.
> ![image](https://user-images.githubusercontent.com/78714518/122026930-49407580-cde8-11eb-8071-3e0d993b8293.png)
> EB: ideally, the table name Products(6) and the Sort button information should be read.
> 
Please ignore the 'List with 6 items' being announced now. This is happening because `listbox` role is applied on the `fd-grid-list` component and the header `Products(6)` is a child of that, so on navigating to `Products(6)`, it announces the list item details too. This will be fixed when structural changes are addressed as part of issue #5655 

> Issue 2:
> Fixed
> 
> issue 3:
> Fixed
> 
> Issue 4:
> ![image](https://user-images.githubusercontent.com/78714518/122028004-37130700-cde9-11eb-8705-0b4593ee3d37.png)
> OB: Additional information is conveyed.
> EB: ideally, the table name Products(7) and the Sort button information should be read.
> 
Same as 1st issue: Please ignore the 'List with 6 items' being announced now. This is happening because `listbox` role is applied on the `fd-grid-list` component and the header `Products(6)` is a child of that, so on navigating to `Products(6)`, it announces the list item details too. This will be fixed when structural changes are addressed as part of issue #5655 

> Issue 5:
> Fixed
> 
> The cards are not navigable by arrow keys.

Issue https://github.com/SAP/fundamental-ngx/issues/5655 is open for keyboard navigation for Grid List. The solution requires structural changes and is not straightforward, so it won't be handled in this PR/issue.


cc:  @puru-hk

**Comment by puru-hk** [comment]
Issues 1, 4 and 5 will be tracking in issue 5655. Hence this issue can be closed

**Comment by Sanjay-1312** [comment]
Hello Colleagues,

As per the above reply this ticket will closed and will be referring the issue 5655 for the issue 1, 4 and 5.

Thanks and Regards
Sanjay
